test/exec/test_io_uring_context.cpp

```diff
@@ -301,14 +301,22 @@ TEST_CASE("io_uring_context schedule_after -1s", "[types][io_uring][schedulers]"
     scope_guard guard{[&]() noexcept {
       context.request_stop();
     }};
-    bool is_called = false;
+    bool is_called_1 = false;
+    bool is_called_2 = false;
+    auto start = std::chrono::steady_clock::now();
     sync_wait(when_any(
       schedule_after(scheduler, -1s) | then([&] {
         CHECK(io_thread.get_id() == std::this_thread::get_id());
-        is_called = true;
+        is_called_1 = true;
       }),
-      schedule_after(scheduler, 5ms)));
-    CHECK(is_called);
+      schedule_after(scheduler, 5ms) | then([&] {
+        is_called_2 = true;
+      })));
+    auto end = std::chrono::steady_clock::now();
+    auto diff = end - start;
+    CHECK(diff < 5ms);
```

Comment:
The failure of this test perhaps indicates that work scheduled to run in the future on the io_uring scheduler doesn't complete promptly when it has been cancelled.

Reply:
Hm, it might be as you say. I will look into it asap. The failing test uses kernel 5.4 and the implementation not touched by this pr (yet) because I only changed the  `IORING_OP_ASYNC_CANCEL` code path, which requires  kernel 5.5 and newer.